Shippo/APIResource.cs

```diff
@@ -433,6 +438,49 @@ public ShippoCollection<Manifest> AllManifests (Hashtable parameters)
 
         #endregion
 
+        #region Batch
+
+        public Batch CreateBatch (String carrierAccount, String servicelevelToken, String labelFiletype,
+                                  String metadata, List<Batch.BatchShipment> batchShipments)
+        {
+            string ep = String.Format ("{0}/batches", api_endpoint);
+            Hashtable parameters = new Hashtable ();
+            parameters.Add ("default_carrier_account", carrierAccount);
+            parameters.Add ("default_servicelevel_token", servicelevelToken);
+            parameters.Add ("label_filetype", labelFiletype);
+            parameters.Add ("metadata", metadata);
+            parameters.Add ("batch_shipments", batchShipments);
+            return DoRequest<Batch> (ep, "POST", serialize(parameters));
+        }
+
+        public Batch RetrieveBatch (String id, Hashtable parameters = null)
+        {
+            string ep = String.Format ("{0}/batches/{1}", api_endpoint, HttpUtility.HtmlEncode(id));
+            if (parameters != null)
+                ep = String.Format ("{0}?{1}", ep, generateURLEncodedFromHashmap (parameters));
+            return DoRequest<Batch> (ep, "GET");
+        }
+
+        public Batch AddShipmentsToBatch (String id, List<Hashtable> shipments)
```

Comment:
Currently, AddShipmentsToBatch requires a list of shipment objects:

{ "shipment" : ... }

It's not set up to be how the API docs currently has it as. (Remember there's an issue and it hasn't been fixed)